### PR TITLE
Bump WP & WC tested up to versions for v3.5.0

### PR DESCRIPTION
**Pull request by achyuthajoy**
#### Changes proposed in this Pull Request

This PR updates the WordPress & WooCommerce tested up to versions.

#### Testing instructions

None.
-------------------

- [x] Added changelog entries to both `readme.txt` and `changelog.txt` (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**
- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

**Review by vbelolapotkov — CHANGES_REQUESTED** [review]
@achyuthajoy version numbers look good but more changes are necessary, see the following examples:

- We'd need to bump min supported WC version as here: https://github.com/Automattic/woocommerce-payments/pull/2263/files
- changelog entry is necessary for this ^
- CI matrix needs update as well (like here: https://github.com/Automattic/woocommerce-payments/pull/2945/files)

File: .github/workflows/compatibility.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
       fail-fast:    false
       max-parallel: 10
       matrix:
-        woocommerce: [ '4.4.0', '4.5.0', '4.6.0', '4.7.0', '4.8.0', '4.9.0', '5.0.0', '5.1.0', '5.2.0', '5.3.0', '5.4.0', '5.5.0', '5.6.0', '5.7.0', '5.8.0', '5.9.0' ]
+        woocommerce: [ '4.4.0', '4.5.0', '4.6.0', '4.7.0', '4.8.0', '4.9.0', '5.0.0', '5.1.0', '5.2.0', '5.3.0', '5.4.0', '5.5.0', '5.6.0', '5.7.0', '5.8.0', '5.9.0', '6.0.0' ]
```
Comment (vbelolapotkov) [inline .github/workflows/compatibility.yml:14]: That list doesn't look good, but is out of scope of this PR.

**Review by vbelolapotkov — APPROVED** [review]
LGTM! :shipit:

File: .github/workflows/compatibility.yml (hunk 1 above)
Comment (achyuthajoy) [reply .github/workflows/compatibility.yml:14]: Yeah. This can be addressed with the new release train process starting January 2022

**Pull request merged**